tools/representative_length.py

```diff
@@ -0,0 +1,103 @@
+from shapely.geometry import MultiLineString, LineString
+from gistools.utils.collection import MemCollection
+
+
+def representative_length(line_col, profile_col):
+    """ creates a MemCollection with a line geometry, attributes copied from the input profile file,
+        and three added fields for the representative length: pre length, post length and total
+        representative length.
+
+        receives two line input files:
+        - line shape representing the waterways
+        - line shape representing the profiles
+
+        returns MemCollection profile_col with the representative length data added
+        """
+
+    # Initialising prof_pk; a unique identifier for each profile
+    prof_pk = 0
+
+    # Loop through each line and select the profiles that intersect the line
+    for feature in line_col:
+        # Initialize the collections
+        points_col = MemCollection(geometry_type='Point')
+        sorted_col = MemCollection(geometry_type='Point')
+
+        # Check if line is LineString or MultiLineString and create the appropiate line object
+        if type(feature['geometry']['coordinates'][0][0]) != tuple:
+            line = LineString(feature['geometry']['coordinates'])
+        else:
+            line = MultiLineString(feature['geometry']['coordinates'])
+
+        # Loop through the profiles within the bounding box of the line
+        for profile in profile_col.filter(bbox=line.bounds, precision=10**-6):
+            if type(profile['geometry']['coordinates'][0][0]) != tuple:
+                prof = LineString(profile['geometry']['coordinates'])
+            else:
+                prof = MultiLineString(profile['geometry']['coordinates'])
+
+            # Making an intersection of the profile with the line creates a point
+            x = line.intersection(prof)
+
+            # If the profile has an intersect with the line, and thus a point is created, a unique prof_pk is assigned
+            # to the profile, distance from the beginning of the line to the point is calculated, and it is added to
+            # the points collection.
+            if x:
+                profile['properties']['prof_pk'] = prof_pk
```

Comment:
dit stukje snap ik niet, waarom wordt er een nieuw uniek prof_pk aangemaakt?

Reply:
Dat heb ik gedaan zodat ik de punten (die gemaakt worden na de intersectie met de lijnen) weer terug gelinkt kunnen worden aan het profiel waar ze bij horen. Later bedacht ik dat het wellicht helemaal niet nodig is om de punten te verzamelen maar dat de informatie rechtstreeks aan het profiel gelinkt kan worden, maar toen had ik het al gecommit. Bovendien weet ik dat niet zeker, ga ik nog even naar kijken.